### PR TITLE
Port to GTK4

### PR DESCRIPTION
**Pull request by sdroege**
This is on top of https://github.com/plotters-rs/plotters-gtk-demo/pull/1

### CODE REVIEW

**Comment by sdroege** [comment]
Also refactoring this a bit now to make it less complicated.

File: src/gaussian_plot/ui.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    <property name="layout-manager">
+      <object class="GtkBinLayout"/>
+    </property>
+    <child>
+      <object class="GtkDrawingArea" id="drawing_area">
+        <property name="vexpand">1</property>
+        <property name="width-request">600</property>
+        <property name="height-request">600</property>
+      </object>
+    </child>
```
Comment (bilelmoussaoui) [inline src/gaussian_plot/ui.xml]: I would drop this and override the snapshot vfunc of GtkWidget

Comment (sdroege) [reply src/gaussian_plot/ui.xml]: Thanks, indeed makes it quite a bit simpler

File: src/window/ui.xml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                  <property name="column">3</property>
+                  <property name="row">2</property>
+                </layout>
+              </object>
+            </child>
+          </object>
+        </child>
+        <child>
+          <object class="GaussianPlot" id="plot">
+            <property name="vexpand">True</property>
```
Comment (bilelmoussaoui) [inline src/window/ui.xml:188]:
```suggestion
            <property name="vexpand">True</property>
```
this should probably be set from the widget itself on constructed or not?

File: src/gaussian_plot/mod.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+use gtk::glib;
+
+mod imp;
+
+glib::wrapper! {
```
Comment (bilelmoussaoui) [inline src/gaussian_plot/mod.rs:5]: I wonder if it wouldn't better to not split those into two files :)

File: src/window/mod.rs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+mod imp;
+
+use gtk::{gio, glib};
+
+glib::wrapper! {
+    pub struct Window(ObjectSubclass<imp::Window>)
+        @extends gtk::ApplicationWindow, gtk::Window, gtk::Widget, @implements gio::ActionMap, gio::ActionGroup;
+}
```
Comment (bilelmoussaoui) [inline src/window/mod.rs]:
```suggestion
        @extends gtk::ApplicationWindow, gtk::Window, gtk::Widget;
}
```

Not really useful here

File: src/window/ui.xml (hunk 2 above)
Comment (sdroege) [reply src/window/ui.xml:188]: Not sure. It seems like a concern of the window to decide how big the plot should be, not of the plot widget itself

File: src/gaussian_plot/mod.rs (hunk 3 above)
Comment (sdroege) [reply src/gaussian_plot/mod.rs:5]: I prefer it like this, seems cleaner to me :)

File: src/window/ui.xml (hunk 2 above)
Comment (sdroege) [reply src/window/ui.xml:188]: Done via the default window size instead, that makes more sense

File: Cargo.toml [hunk 5]
```diff
@@ -7,14 +7,14 @@ edition = "2018"
...
 
 [dependencies.plotters]
 git = "https://github.com/plotters-rs/plotters"
 default_features = false
 features = ["surface_series"]
 
 [dependencies.plotters-cairo]
-git = "https://github.com/plotters-rs/plotters-cairo"
+git = "https://github.com/sdroege/plotters-cairo"
+branch = "cairo-rs-0.16"
```
Comment (sdroege) [inline Cargo.toml]: This needs https://github.com/plotters-rs/plotters-cairo/pull/26 merged first

**Comment by sdroege** [comment]
@AaronErhardt This should be ready for merging now.

**Review by AaronErhardt — APPROVED** [review]
Looks good, but sadly I don't have write access to this repo. 

@38 could you please grant me the permission? Also, it'd be great if others in the plotters organization could publish crates.

**Review by AaronErhardt — APPROVED** [review]

**Pull request merged**